### ISSUE TITLE
Move `Diagnostics` from `ConsoleDriver` to `View`

### ISSUE THREAD

**Issue by tig** [gui-cs/Terminal.Gui#3304]
The two diagnostic flags we have now:

```cs
public abstract class ConsoleDriver
{
    /// <summary>Enables diagnostic functions</summary>
    [Flags]
    public enum DiagnosticFlags : uint
    {
        /// <summary>All diagnostics off</summary>
        Off = 0b_0000_0000,

        /// <summary>
        ///     When enabled, <see cref="View.OnDrawAdornments"/> will draw a ruler in the frame for any side with a padding
        ///     value greater than 0.
        /// </summary>
        FrameRuler = 0b_0000_0001,

        /// <summary>
        ///     When enabled, <see cref="View.OnDrawAdornments"/> will draw a 'L', 'R', 'T', and 'B' when clearing
        ///     <see cref="Thickness"/>'s instead of ' '.
        /// </summary>
        FramePadding = 0b_0000_0010
    }
```

are 

a) on `ConsoleDriver`
b) implemented by `Adornment`

I want a 3rd, that I've put in #3273:

```cs
        /// <summary>
        ///     When enabled, <see cref="Adornment.OnMouseEnter(Gui.MouseEvent)"/> and <see cref="Adornment.OnMouseLeave(Gui.MouseEvent)"/>
        ///     will invert the foreground and background colors.        ///     
        /// </summary>
        HighlightAdornmentOnMouseEnter = 0b_0000_00100
```

These should be moved out of `ConsoleDriver` and refactored to be on `View`.

**Comment by dodexahedron** [comment]
A couple of suggestions that are pretty low-impact to do:

 - (Opinion): I'd suggest a different name for the enum, as the current name suggests an intent related to logging.
 - (Best practices/design guidelines): Don't make it a sub-type of any other type. Classes aren't namespaces.

**Comment by dodexahedron** [comment]
### Follow-up on the whole sub-type/namespace thing:

That's one of the worst/biggest "weird" things about the current state of the code, all else being equal. The standard and a pretty major expectation for users is that namespaces are used where and when appropriate. Typically, if it makes sense to put a code file in a folder, it makes sense for it to be in a namespace named for that folder. Visual Studio even strongly suggests that out of the box.

It makes no sense at all for types with different applicability to be in the same namespace, particularly if they're publicly visible types, unless a project is significantly smaller than this library.

And, with things like global usings, it is completely transparent anyway, so places no burden on us, while providing an expected basic organization for consumers.

### Follow-up on not using child/contained types:
First reason is it's against recommendations to ever have one that is public and intended to be used.

But it also creates problems, especially when generics come into play. A generic type's sub-types are unique to each run-time constructed type, meaning they are not the same, even if the sub-type is not generic itself. It is a _part_ of the enclosing type, period, in all situations, and you pay for it whether it is used or not, too.

The whole generic thing becomes really nasty when you try to pass something around with a base class or interface that in any way needs to interact with those sub-types. It may work at compile-time but, especially if the type isn't sealed, problems can and are likely to occur at run-time, as soon as a constructed type that doesn't actually work with the assumptions made in that base type or interface.

Static analysis is capable of predicting a very small subset of the possible problems and will warn about it when it sees them, but the first time something goes through a cast or a run-time type check, either a runtime error or unexpected behavior will result, due to the type not being identity-convertible (cast situations) or not an exact match (type checks such as `is sometime`).

Type parameter constraints on generics do help to the extent that they are actually respected by the implementation, but it's best just to avoid the situation entirely and never do it for anything publicly visible, period, except on sealed types.

**Comment by tig** [comment]
> ### Follow-up on the whole sub-type/namespace thing:

I'm confused why you're woried about this. We are already in 100% alignment and we got rid of all nested types (except, obviously this one, which I fix in #3305.

**Comment by tig** [comment]
> ### Follow-up on the whole sub-type/namespace thing:
> That's one of the worst/biggest "weird" things about the current state of the code...

We've gone back and forth on this. The reason I pushed to have just `Terminal.Gui` was for how the API was exposed in the API documentation. IIRC (and it's been a while), when we had a bunch of namespaces the API documentation was very hard to navigate. Now it's a nice clean flat API. 

That said, i'm open to revisiting this.
